### PR TITLE
Feature/9869 new nationwide incidences with backward compatibility

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add the The nationwide cards sor the 7-day incidence of new infections are combined with the 7-day incidence for hospitalization according to Figma

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9869

## Screenshots
Simulator Screen Shot - iPhone 11 - 2021-10-11 at 02 32 40

### CODE REVIEW

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift [hunk 1]
```diff
@@ -47,11 +51,36 @@ class HomeStatisticsCellModel {
...
-
+	private var isCombinedIncidenceCardSupported: Bool {
+		#if DEBUG
+		if isUITesting {
+			return !LaunchArguments.statistics.useMockedOldAppVersion.boolValue
+		} else if isOldAppVersion {
+			return false
+		}
+		#endif
+		// we show the combined card 10 only for 2.13 or later, else we show 2 and 8
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift:76]: Won't the version for this delivery will always the 2.13? In which scenario can we end up with this code and an app version < 2.12?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 2]
```diff
@@ -86,6 +86,9 @@ class HomeStatisticsTableViewCell: UITableViewCell {
 	) {
 		guard cellModel == nil else { return }
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:87]: What is about cell reusing? Cells can be reused and configured for another indexpath. Then this configure will be called with data for the corresponding (new) indexpath. But it won't be reconfigured? Is cell reusing somehow disabled?

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift]: I just updated the code

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift [hunk 3]
```diff
@@ -47,11 +51,36 @@ class HomeStatisticsCellModel {
 		localStatisticsProvider.remove(region)
 	}
 
+	func filterKeyFigures(keyFigures: [SAP_Internal_Stats_KeyFigureCard]) -> [SAP_Internal_Stats_KeyFigureCard] {
+		if isCombinedIncidenceCardSupported {
+			return keyFigures.filter { $0.header.cardID != 2 && $0.header.cardID != 8 }
+		} else {
+			return keyFigures.filter { $0.header.cardID != 10 }
+		}
+	}
```
[the remark this answers is not in the document]
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift]: thanks!, I just updated the code

File: src/xcode/ENA/ENAUITests/LaunchArguments.swift [hunk 4]
```diff
@@ -94,8 +94,8 @@ enum LaunchArguments {
 		}
 	}
 	enum statistics {
-		/// To use the mock data for statistics
-		static let useMockDataForStatistics = LaunchArgument(name: "useMockDataForStatistics")
+		/// To use older app version without the combined nationwide statistics card
+		static let useMockedOldAppVersion = LaunchArgument(name: "useMockedOldAppVersion")
 	}
```
Comment (nickguendling) [inline src/xcode/ENA/ENAUITests/LaunchArguments.swift:99]: Seems like this whole enum can be deleted now

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 5]
```diff
@@ -1468,7 +1468,15 @@ enum AppStrings {
...
 			}
-			
+			enum Combined7DaysIncidence {
+				static let title = NSLocalizedString("Statistics_Card_CombinedIncidence_Title", comment: "")
+				static let today = NSLocalizedString("Statistics_Card_CombinedIncidence_Today", comment: "")
+				static let yesterday = NSLocalizedString("Statistics_Card_CombinedIncidence_Yesterday", comment: "")
+				static let date = NSLocalizedString("Statistics_Card_CombinedIncidence_Date", comment: "")
+				static let primaryLabelSubtitle = NSLocalizedString("Statistics_Card_CombinedIncidence_PrimaryLabelSubtitle", comment: "")
+				static let secondaryLabelSubtitle = NSLocalizedString("Statistics_Card_CombinedIncidence_SecondaryLabelSubtitle", comment: "")
+			}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1478]: You can delete the old two cards from the AppStrings as well as the Localizable.strings

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 6]
```diff
@@ -539,6 +539,10 @@ enum AccessibilityIdentifiers {
 			static let title = "AppStrings.Statistics.Card.IntensiveCare.title"
 			static let infoButton = "AppStrings.Statistics.Card.IntensiveCare.infoButton"
 		}
+		enum Combined7DayIncidence {
+			static let title = "AppStrings.Statistics.Card.Combined7DayIncidence.title"
+			static let infoButton = "AppStrings.Statistics.Card.Combined7DayIncidence.infoButton"
+		}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:545]: Old two cards can be deleted here as well, and the illustrations from the asset catalogue

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift (hunk 6 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:539]: done

**Review by ArturFriesen — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**